addons/func_godot/src/map/func_godot_map.gd

```diff
@@ -67,6 +77,9 @@ var entity_collision_shapes: Array = []
 
 # Overrides
 func _ready() -> void:
+	if Engine.is_editor_hint() && auto_build:
+		verify_and_build()
```

Comment:
This will build every single time this scene is opened in the editor, which means you'll be hit with a lag spike every time you open a map as it builds. Can we do something else here?

Reply:
If you'd like, but that's part of the point for me. If you've got a particularly large map that you don't want to generate every time you open it, you can simply turn the bool off for it. 

Without getting far more complicated with it, there's no way to check if a Resource *has been* modified since the last time a Scene was open, and so simply rebuilding each time it's opened seemed minor to me in favour of it being up to date. I was impressed with how performant the plugin has been since I started using it, but I have only used extremely simple maps. 

On a side note, this PR *doesn't* deal with the requirement to open each Scene containing a map file you've modified each time you alter them. Batch editing maps seems like it could grow to be a huge pain, but I don't know a solution personally. You'd need to find map Nodes and load the Scene up in the background before re-saving the file each time the Resource is edited, which sounds a little sketchy. Funnily enough, it'd also solve the problem listed here :P

Reply:
Perhaps it should be moved from a func_godot_map field, to a func_godot_map_settings field, this way it applies to all the maps using these settings

Reply:
I considered that, the trouble is that it's far more per-map considerate than any other setting in that Resource.

Reply:
Perhaps... perhaps we can have one in the map settings that if set, overrides it for all maps using it as a setting? Ultimately, I'm not too fussed about it

Reply:
I had thought about suggesting it be in map settings, but honestly I feel it fits more on a per-map basis rather than a gamewide basis. I'd have to agree with Psprite's judgment in this specific case.

Reply:
I'm not 100% convinced. Currently a lot of the settings that you could argue are per-map, are in these settings. Stuff like unshaded, save generated materials, use trenchbroom groups hierarchy etc. It also makes an opinion that map settings are gamewide, which I also don't think is always the case(unshaded is a good example of this!).
- You want to use dev settings for development("mat_fullbright 1" for unshaded) and production settings for prod
- Reloading only props that are being worked on(dev/prod idea, but with prod you don't want runtime map loading in your finished exports. Not everyone wants to expose dev tools in their finished games)
- Toggle on for the map setting, open all scenes using these map settings to regenerate them, and then toggle it off in the settings(as opposed to turning it on and off in each FuncGodotMap scene if they aren't using a premade FuncGodotMap scene). If you have to turn it on manually for all maps, that will get tedious fast

While I agree it's more per-map specific, so are others like unshaded and use trenchbroom group hierarchy

Reply:
Fair assessment. The original intent of a map settings resource was to share setups across maps without having to re-do them every time. A side benefit of this was using the same map with different settings could yield different results, and a way to apply it to the auto_rebuild / auto_uv_unwrap features would be to have a "production map settings" and a "release map settings", or some other combination that you can hot swap in and out.

While you could have an additional "override" variable that dictates whether an auto rebuild occurs or not, I'd lean on the side of "not". While we do have some overrides scattered about here and there, we got rid of most of the ones that were found in Qodot as they were bandaids to avoid breaking projects.

I'd have to defer to @RisingThumb in this case, the auto rebuild setting does belong in the map settings resource given what it's used for already. Shouldn't be all that difficult to change over though.